### DISCUSSION TITLE
Send HTML email (from directus templates) in flows ?

### DISCUSSION THREAD

**Discussion by washaweb** [directus/directus#15639] [Archive]
Hello community, 

Thanks you all for this amazing work you do with Directus, I love it !

I'm building a contact form using Directus flows: I've got a `contactform` collection to store my forms in Directus. 
So, for now, when a new form is submitted (using an API call), a flow is triggered on `contactform`:`items.create`, it sends an email using `$trigger.payload`. Everything works well...

Now I would like to send rich HTML emails. 

Using flows, I wonder if there is a way to get access to directus email templates from flows ?

As for now the email body is only editable with a markdown editor.  I've tryed to put some HMTL code in the body (via the wysiwyg editor) but it seems to be trimmed when email is send.

So, I'm I doing well ? Is there another way to achieve this ? I Need some advice here, please.

If not, I think it would be great to add this feature in flows... just an idea !

**Comment by igorazevedo** [comment]
You can use the "Run script" operation to generate your HTML, and then use the result of that as the body for the Send Email operation.

I've tested it and it works, although I'm not sure if this is the intended behavior or if they just forgot to sanitize the variables set in the email's body.

**Reply by washaweb** [reply]
I've tryied it and does'nt work, or maybe I did something wrong... 

Searching forward in the Q/A list, I've found this answer, so I think this feature is still in discussion at the moment :
https://github.com/directus/directus/discussions/14091

I also decided to bypass my Flows using a hook instead, I share my final code here for people who don't want to search further for a solution, hope that it helps.

File `extensions/hooks/contactform/index.js` <-- replace the hook name `contactform` with the name of your collection ;-)

```js
module.exports = function registerHook({ action }, { services, exceptions }) {
  const { MailService } = services
  const { ServiceUnavailableException, ForbiddenException } = exceptions

  // Send mail to admin when contact form is created
  action('items.create', async ({ collection, payload }, { schema }) => {
    if (collection !== 'contactform') return
		
    const mailService = new MailService({ schema });

    try {
      await mailService.send({
        to: process.env.ADMIN_EMAIL, //you need to define this variable in your .env (ex: ADMIN_EMAIL=your@email.com)
        subject: 'New contact form sent',
        text: `Contact form from ${payload.first_name} ${payload.last_name}`,
        template: {
          name: 'contact-form-admin',
          data: {
		payload
          },
        },
      })
			
    } catch (error) {
      console.error(error)
      throw new ServiceUnavailableException(error)
    }
    return
  })
}
```
You need to create the corresponding template in `extensions/templates/contact-form-admin.liquid`

```liquid
{% layout "base" %}
{% block content %}
<p>Hello,br /><br />
A new contact form has been sent :</p>
<ul>
<li>contact: {{ payload.first_name }} {{ payload.last_name }}</li>
<li>email: {{ payload.email }}</li>
<li>title: {{ payload.title }}</li>
<li>message: {{ payload.message }}</li>
</ul>
<p>The webmaster</p>
{% endblock %}
```

**Reply by igorazevedo** [reply]
Which version are you on?

I created a "Run script" operation with the key "generate_content" and the following code:

```javascript
module.exports = async function(data) {
	return { body: '<p>This is a <strong>test</strong>!</p>'};
}
```

And then a "Send Email" operation where I added to the body the following:

```
{{generate_content.body}}
```

I was able to send the email with the HTML correctly formatted on 9.17.4. But then again, as I said, I don't know if this is the expected behavior. Your solution is probably cleaner.

**Reply by washaweb** [reply]
ok, I was doing it wrong (in Flows).

I now have found another solution, with hooks. But I keep your code example as a reference for other Flow's actions I need to do.

thank you !

**Reply by washaweb** [reply]
By the way, I use Directus `9.16.1`. I'll upgrade to `9.18.0` to test your solution.
